tfmigrate/state_replace_provider_action.go

```diff
@@ -0,0 +1,37 @@
+package tfmigrate
+
+import (
+	"context"
+
+	"github.com/minamijoyo/tfmigrate/tfexec"
+)
+
+// StateReplaceProviderAction implements the StateAction interface.
+// StateReplaceProviderAction replaces a provider from source address to destination address in
+// the same tfstate file.
+type StateReplaceProviderAction struct {
+	// source is the provider address to be replaced.
+	source string
+	// destination is the new provider address.
+	destination string
+}
+
+var _ StateAction = (*StateMvAction)(nil)
```

Comment:
It should be StateReplaceProviderAction, not StateMvAction, since this is the idiomatic expression for type-checking that the struct implements the interface.